### PR TITLE
Improve completionComplexEdit command

### PR DESCRIPTION
**Pull request by etvorun**
We need to open a document if it is not opened yet when handling complex edit like adding a new event handler to MyPage.xaml.cs after selecting "add new event handler" in MyPage.xaml completion popup.

### CODE REVIEW

File: src/lsptoolshost/commands.ts [hunk 1]
```diff
@@ -103,73 +103,70 @@ async function completionComplexEdit(
...
+    }
 
-            if (success) {
-                break;
-            }
+    // Use editor if we need to deal with selection or snippets.
+    let editor: vscode.TextEditor | undefined = undefined;
+    if (isSnippetString || newOffset >= 0) {
+        editor = await vscode.window.showTextDocument(document);
+        if (editor == undefined) {
```
Comment (dibarbet) [inline src/lsptoolshost/commands.ts:118]:
```suggestion
        if (editor === undefined) {
```

File: src/lsptoolshost/commands.ts [hunk 2]
```diff
@@ -103,73 +103,70 @@ async function completionComplexEdit(
...
-        outputChannel.show();
-        outputChannel.appendLine(`${componentName} ${errorMessage}`);
-
-        if (editor === undefined) {
-            outputChannel.appendLine(
-                `${componentName} Can't find visible document with uri.fsPath: '${uri.fsPath}' and uri.path: '${uri.path}'`
-            );
-
-            outputChannel.appendLine(`${componentName} URIs of all visible documents:`);
-            for (const visibleEditor of vscode.window.visibleTextEditors) {
```
Comment (dibarbet) [inline src/lsptoolshost/commands.ts:160]: @genlu do you think we still need to keep this logging or can it be removed?

Comment (etvorun) [reply src/lsptoolshost/commands.ts:160]: I found logging useless. I can remove it you also see no value here.

Comment (genlu) [reply src/lsptoolshost/commands.ts:160]: Those were added when we tried to figure out the uri issue that broke complex edit. If we haven't seen any such issues recently, then I'm fine removing them

Comment (etvorun) [reply src/lsptoolshost/commands.ts:160]: Logging removed

Comment (dibarbet) [reply src/lsptoolshost/commands.ts:160]: we should keep at least the logging we had when we were not able to apply the edits.  throwing an error doesn't always bubble up to a visible location.

My original comment was asking Gen about the logging of the visible editor stuff that was also removed here.  It sounds like this part is OK to remove.

Switching it back to how you originally had it and it should be good to go in.

**Review by dibarbet — CHANGES_REQUESTED** [review]

**Review by dibarbet — APPROVED** [review]

**Pull request merged**